### PR TITLE
Economy 2019 Part C Branch B

### PR DESCRIPTION
**Pull request by Kurfursten**
The main economy reform was too vulnerable to blocking because it lacked atomicity. I'm going to split it in three to make it easier to push through. Part A is aimed at accounts issues, Part B will target vending machine improvements, and Part C will be where we take on the miner metaclub.

Branch B represents the "ideal timeline". I've removed the highly controversial previous name for points, now IDs can get _labor points_. You can still hold up your ID to show off labor points. There have been no changes to mining vend prices or mineral values, so miner progression is unaltered and no worries about people breaking into cargo for cheap stuff at the mining vend.

🆑 
* tweak: Miners now receive labor points on their ID from the smelter. Mining vendor items all cost labor points.

### CODE REVIEW

**Comment by ghost** [comment]
Where are research points, plant points, cook points and crime points?

More currency types will lock content to specific jobs without means of bypass.

**Comment by Kurfursten** [comment]
If we come to a point in time where someone adds labor points for other jobs we could answer that. There are a multitude of options: you could convert labor points into a list, you could just check that they have labor points and the associated access (accepting that it's okay for someone to buy miner goodies if they spent their shift getting points cooking), etc. At this point it's not important to waste too much thought on since no one has implemented those. Not only would you need to come up with a way of valuing labor (easier for some jobs like cook than officer or surgeon) but you would need to have a vendor full of goodies and toys.

**Comment by DrSnips** [comment]
I wouldn't say it job locks the items since the mining vendor is most often vomiting it's contents on the floor for free at an unlucky mouse. I'd much prefer this way over method A due to the fact that it permanently solves the economy inflation issue. I can guarantee you that miners will bitch to no end that the amount of currency they are getting from A will be "too little" dispite the economy rework.

**Comment by DrSnips** [comment]
Something I would like to point out is the lack of an explanation why they dislike this version. 11 nos but no actual reasoning behind it.

**Comment by SonixApache** [comment]
I'll explain my :-1:
My two issues were
1: I can't take things from the mining vendor when I want them (I often go and buy soap from cargo as non-cargo, I enjoy getting a knife holster as cargo)
You could argue that I could just barter with a miner for it, but what self-respecting miner would actually waste their jetpack / kinacc / pokemans money on buying me stuff in exchange for useless money?
and the second point, which is closely related;
2: Miners wouldn't be able to use the resource they acquire by doing their job on anything but more job things, which is pretty lame. You COULD barter your miningbux (in other words, your time) against money, but I feel like the incentive for that would be really low.

**Comment by Kurfursten** [comment]
If people are willing to break into Cargo to buy soap and knife holsters (both free from botany with a little human interaction) I've got a grim feeling about PKAs and Jetpacks in a future where they are within purchase range.

**Comment by bathosbathtime** [comment]
The original name still leaves a bad taste in my mouth, and I don't think we should have an additional currency in the game when coins already feel redundant (miners also have piss easy access to coins).
The whole thing feels like adding convoluted systems to something already too convoluted.

**Comment by ghost** [comment]
Why is breaking into a place even an argument here? Anyone can break anywhere and steal anything. Why is a mining vendor special?

**Comment by Kurfursten** [comment]
Mining gear is inside a vendor. If you want to lift glubbs from engineering, for example, that requires either you have access to an emitter, an emag, attracting bhangometer attention with an IED, or engineering deserves it because they left a locker unlocked like idiots. As Sonix points out, anyone can just buy up the PKAs at the miningvend which is a low barrier to entry. Putting it behind labor points solves this problem, and so does keeping things like PKAs at a high price (because, honestly, no one really cares if sonix is buying holsters and soap) - the labor or price acts as the barrier.

@bathosbathtime If coins are an issue for you, let's get rid of coins in machines. They were always the more ill conceived "currency" anyway and have even less rhyme or reason to them.

**Comment by agentofzog2** [comment]
Let's go over what's in the minding vend without looking at the prices too much:
canary: for decorative purposes only
Chili: only for when you don't want to walk to the kitchen or a food vend, could have just ordered pizza instead
knife holster: mildly useful, can get from other places
soap: dude slips, can get from other places
facehugger: sometimes funny
trainer's belt: useful for pokeman
lazarus capsule: pokemen
Lazarus injector: pokemen
sonic jackhammer: only useful if someone buys it roundstart, (this is currently not happening since it'll eat cargo's entire account)
mining drone: can either drag it or the ore box so it's not that great. Useful for simple mob invasions
wormhole jaunter: makes a wormhole, dubious use
resonator: I haven't used it in forever but it's like a really unwieldy melee range weapon that can also dig through rocks if for some reason you have no Ddrill
PKA: useful for shooting roid monsters, pretty weak compared to other guns
jetpack: useful, can be taken  from other places
hookshot: fun
advanced lazarus injector: pokeman
Whiskey: it's whiskey
Special Picks (coin): see jackhammer
Cyborg Hookshot: Fun but supply borgs tend to usually switch modules after they're done mining if they appear at all.

The actual essential items are the PKAs (if you a good boi) of which you'd buy several and the jetpack, then you have the pokemon items of which you want a lot of but pokemon are hard to use as nonantag due to the collateral damage. The rest are stuff that's just4fun and probably not worth buying if you actually have to work for it, kinda crap or highly situational. There's a reason why most of the stuff is just left laying on the floor in the current mouse and shoot wire meta.


tl:dr: The mining vend is full of crap and only a few items are actually useful to miners who just want to do their job, restricting miners to only being able to use their roid dollars on it is a shit idea especially when the current economy has no other way of generating money for that 20 shard autism engine and getting traders to stand on this rune for just a moment.

**Comment by Kurfursten** [comment]
I don't think your point connects to what your argument. It seems like you're arguing *for* this PR, basically.

>Miners won't be making anything but roid dollars anyway
>Therefore vendor should just cost roid dollars, not regular dollars

**Comment by agentofzog2** [comment]
Miner can currently buy: 2 useful things from vend and loads of useful things from other places
Miner with roidcoins can  buy: 2 useful things from vend

Also this would ruin the mining based Permas since sec can't profit off of the prison mining complex.

**Comment by Kurfursten** [comment]
They already can't, it requires mining access to claim money from the smelter.

Miners buying things from other places with minerbux is the problem (inflation). No system of exchange makes sense when miners aren't in the same galaxy in terms of income. Put another way, I approximate that the average crewman values $50 about the same as a miner would value $40,000.

Miners already get their regular wages like anyone else too (not that they notice)

**Comment by ghost** [comment]
>it requires mining access to claim money from the smelter.

It does not, anyone can slap the console with their ID to insert it and claim money. I hope this is intended because the alternative makes no sense. Why should only miners be able to claim credits?

**Comment by Kurfursten** [comment]
There's an allowed() check for inserting your ID, so it's indeed bugged if that's not working.

Anyway I think at this point the labor points gang is going to move on to a Branch C proposal in which we bring the smelter in line with the biogenerator, so I'll close this one.
